core/components/header/_actions/search.ts

```diff
@@ -0,0 +1,14 @@
+'use server';
+
+import { z } from 'zod';
+
+import { redirect } from '~/i18n/routing';
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export const search = async (formData: FormData) => {
+  const searchTerm = z.string().parse(formData.get('searchTerm'));
+
+  if (searchTerm) {
+    redirect(`/search?term=${searchTerm}`);
+  }
+};
```

Comment:
Seems like server actions need to be async functions, but there's no async function inside this. Is this the best way to solve this?

Reply:
Do we need this to be a server action? Could it be a client side `router.push`?

Reply:
Pushed an update to just use the client.